### PR TITLE
openstack: Add e2e job for cluster-api-provider-openstack

### PR DESCRIPTION
**Pull request by trown**
Adds a job which runs the openstack-e2e job against CAPO changes.

### CODE REVIEW

**Comment by trown** [comment]
/hold cancel
this job will run max twice a day, not really a big deal to bump the memory

File: ci-operator/jobs/openshift/cluster-api-provider-openstack/openshift-cluster-api-provider-openstack-master-presubmits.yaml [hunk 1]
```diff
@@ -1,5 +1,61 @@
 presubmits:
   openshift/cluster-api-provider-openstack:
+  - agent: kubernetes
+    always_run: false
+    branches:
+    - master
+    context: ci/prow/e2e-openstack
+    decorate: true
+    decoration_config:
```
Comment (mandre) [inline ci-operator/jobs/openshift/cluster-api-provider-openstack/openshift-cluster-api-provider-openstack-master-presubmits.yaml]: This needs to be:

```
    decoration_config:
      grace_period: 0s
      skip_cloning: true
      timeout: 0s
```

**Comment by petr-muller** [comment]
/lgtm cancel

This needs to be regenerated with latest prowgen image

**Pull request merged**